src/tendencies_parametrizations.jl

```diff
@@ -1,23 +1,40 @@
 """
-Compute physical parametrization tendencies.
+    parametrization_tendencies!(diagn::DiagnosticVariables,
+                                M::PrimitiveEquationModel)
+
+Compute tendencies for u,v,temp,humid from physical parametrizations.
+Extract for each vertical atmospheric column the prognostic variables
+(stored in `diagn` as they are grid-point transformed), loop over all
+grid-points, compute all parametrizations on a single-column basis,
+then write the tendencies back into a horizontal field of tendencies.
 """
-function parametrization_tendencies!(
-    Prog::PrognosticVariables{NF},
-    Diag::DiagnosticVariables{NF},
-    M,
-) where {NF<:AbstractFloat}
-    @unpack pres_grid = Diag.grid_variables
+function parametrization_tendencies!(   diagn::DiagnosticVariables{NF},
+                                        M::PrimitiveEquationModel,
+                                        ) where NF
+
+    G = M.geometry
+    column = ColumnVariables{NF}(nlev=diagn.nlev)
+
+    for ij in eachgridpoint(diagn)
+
+        # extract a single atmospheric column for contiguous memory access
+        get_column!(column,diagn,ij,G)
 
-    # The prognostic variable pres has units of log(hPa), whereas the grid-point pressure
-    # field which we require for the parametrizations has units of hPa, so we take the
-    # exponential here.
-    @. pres_grid = exp(pres_grid)
+        fill!(column.u_tend,0)      # set tendencies to 0 to accumulate +=
+        fill!(column.v_tend,0)      # parametrizations into
+        fill!(column.temp_tend,0)
+        fill!(column.humid_tend,0)
```

Comment:
we may want to add more zero fills here to guarantee that reusing `column` for the next grid point doesn't cause issues. E.g. at the moment we set `precip_large_scale` to zero in `large_scale_condensation!`, maybe it makes sense to define a single point were the state of `column` is reset in order to be reused for the next grid point

Reply:
We have `reset_column!` now that does that.